File: Word-Formed/src/com/google/corrigan/owen/wordformed/HiScoreActivity.java
package com.google.corrigan.owen.wordformed;

import java.util.List;

import android.app.ListActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.WindowManager;

public class HiScoreActivity extends ListActivity {
    private HiScoreDataSource datasource;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.hiscore);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        datasource = new HiScoreDataSource(this);
        datasource.open();

        Log.d("HISCORES", "In HiScoreActivity, reading all scores");
        List<HiScore> values = datasource.getAllHiScores();

        // Use the SimpleCursorAdapter to show the
        // elements in a ListView
        HiScoreAdapter adapter = new HiScoreAdapter(this,
                values);
        setListAdapter(adapter);
    }

    @Override
    protected void onResume() {
        datasource.open();
        super.onResume();
    }

    @Override
    protected void onPause() {
        datasource.close();
        super.onPause();
    }

}

File: app/src/main/java/com/google/corrigan/owen/wordformed/HiScoreActivity.kt
package com.google.corrigan.owen.wordformed

import android.os.Bundle
import android.util.Log
import android.view.WindowManager
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class HiScoreActivity : AppCompatActivity() {
    private lateinit var listView: ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.hiscore)
        
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )

        listView = findViewById(android.R.id.list)

        loadHighScores()
    }

    private fun loadHighScores() {
        lifecycleScope.launch {
            Log.d("HISCORES", "In HiScoreActivity, reading all scores")
            val scores = withContext(Dispatchers.IO) {
                AppDatabase.getDatabase(this@HiScoreActivity).hiScoreDao().getAllHiScores()
            }
            
            val adapter = HiScoreAdapter(this@HiScoreActivity, scores)
            listView.adapter = adapter
        }
    }
}
